hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/NearCacheDataRecordStore.java

```diff
@@ -79,6 +81,10 @@ protected NearCacheDataRecord valueToRecord(V value) {
 
     @Override
     protected V recordToValue(NearCacheDataRecord record) {
+        if (record.getValue() == null) {
+            nearCacheStats.incrementMisses();
+            return (V) NULL_OBJECT;
```

Comment:
Also, maybe `NearCache` can take a parameter to its constuctor like `nullValueAllowed` and its default value is `false`. So `nullValueAllowed` decides `null` value is allowed and converts it to `NULL_OBJECT`, otherwise, throws `InvalidParameterException`.